test/CadlRanchProjectsNonAzure.Tests/model/inheritance/RecursiveTests.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using AutoRest.TestServer.Tests.Infrastructure;
+using NUnit.Framework;
+using Scm._Type.Model.Inheritance.Recursive;
+using Scm._Type.Model.Inheritance.Recursive.Models;
+using Azure.Core;
+using System.Linq;
+
+namespace CadlRanchProjectsNonAzure.Tests
+{
+    public class RecursiveTests : CadlRanchNonAzureTestBase
+    {
+        [Test]
+        public Task Type_Model_Inheritance_Recursive_get() => Test(async (host) =>
+        {
+            var response = await new RecursiveClient(host, null).GetRecursiveAsync();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            Assert.AreEqual(0, response.Value.Level);
+            Assert.AreEqual(2, response.Value.Extension.Count);
+            Assert.AreEqual(1, response.Value.Extension[0].Level);
+            Assert.AreEqual(1, response.Value.Extension[0].Extension.Count);
+            Assert.AreEqual(2, response.Value.Extension[0].Extension[0].Level);
+            Assert.AreEqual(1, response.Value.Extension[1].Level);
+            Assert.AreEqual(0, response.Value.Extension[0].Extension[0].Extension.Count);
+            Assert.AreEqual(false, !(response.Value.Extension[1].Extension is ChangeTrackingList<Extension> changeTrackingList && changeTrackingList.IsUndefined));
```

Comment:
let us change this to `Optional.IsCollectionDefined`

Reply:
fixed